[421] Alert when pipeline fails

saliceti opened the pull request:
## Context
Build workflows fail silently

## Changes proposed in this pull request
Notify slack on build, lint, test, deployment failures

## Guidance to review
Compare with teaching vacancies set-up
Make the build fail
Example notification: https://ukgovernmentdfe.slack.com/archives/CN1MCQCHZ/p1637065688221200

## Link to Trello card
https://trello.com/c/8izJtAP0

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

saliceti commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Alert when pipeline failed](https://trello.com/c/8izJtAP0/421-alert-when-pipeline-failed)

saliceti commented:
TODO: Fetch SLACK_WEBHOOK from keyvault instead of Github secrets

cwrw reviewed (commented):
Discussed in person: we would like to alert only on main branches rather than all PRs as there will be a lot of noise otherwise on the channel, since PRs do fail quite often.

saliceti commented:
> Discussed in person: we would like to alert only on main branches rather than all PRs as there will be a lot of noise otherwise on the channel, since PRs do fail quite often.

Fixed.
Also added notification to the DB backup job

cwrw commented on .github/workflows/deploy.yml:
```diff
@@ -63,7 +74,6 @@ jobs:
       - name: Set Environment variable
         run: |
           echo "DOCKER_IMAGE=$DOCKER_IMAGE" >> $GITHUB_ENV
-          if [ "${{ matrix.environment }}" = "production" ]; then echo "SPACE_SUFFIX=prod" >> $GITHUB_ENV; fi;
```
why did we remove the `SPACE_SUFFIX` from being added here?

saliceti replied:
It's a leftover from a previous refactor, SPACE_SUFFIX is not used anymore. I'll move it to a separate commit for clarity.

cwrw commented on .github/workflows/build.yml:
```diff
@@ -228,9 +252,20 @@ jobs:
             }
 
+      - name: Notify Slack channel on job failure
+        if: failure() && github.event_name == 'push'
```
does the `github.event_name == 'push'` what is ensuring it's on main?

saliceti replied:
The workflow is only triggered on pushes to main: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/.github/workflows/build.yml#L4-L6
So we only need a condition on push vs pull_request

cwrw reviewed (commented):
looks good, thank @saliceti just a few questions otherwise 👍
